common-jvm/src/main/java/org/ergoplatform/SigningSecrets.kt

```diff
@@ -0,0 +1,47 @@
+package org.ergoplatform
+
+import com.google.gson.Gson
+import com.google.gson.JsonObject
+import com.google.gson.JsonParser
+import org.ergoplatform.appkit.SecretString
+
+private const val KEY_MNEMONIC = "mnemonic"
+private const val KEY_DEPRECATED_DERIVATION = "pre1627"
+
+data class SigningSecrets(
+    val mnemonic: SecretString,
+    val deprecatedDerivation: Boolean,
+    val password: SecretString = SecretString.empty()
+) {
+
+    constructor(
+        mnemonic: String,
+        deprecatedDerivation: Boolean
+    ) : this(SecretString.create(mnemonic), deprecatedDerivation)
+
+    fun toJson(): String {
+        val gson = Gson()
+        val root = JsonObject()
+        root.addProperty(KEY_MNEMONIC, mnemonic.toStringUnsecure())
+        // TODO BIP-32 fix root.addProperty(KEY_DEPRECATED_DERIVATION, deprecatedDerivation)
+        return gson.toJson(root)
+    }
```

Comment:
I understand that, and the returned json is encrypted immediately (see usages of the method). The encryption itself is depending of platform and user choice and moving it into SigningSecrets class would be possible only by using a new lambda param for the method which, in my opinion, only adds some noise and is harder to read without giving any benefit here.
In case I misunderstood and you were talking about having the whole json and not certain fields encrypted is a bad idea: the idea here was that the serialized secrets should be encrypted as a whole so that no information what fields exactly are set can be read without decrypting. Originally I thought about the mnemonic password here (although it is not implemented in the app).

Reply:
The main issue I see is that the lifetimes of secrets are  not well defined and they leak to GC. It is not a big deal to do whatever necessary with the secrets as long as they don't leak to GC and/or external storage.
The usage of secrets should be similar to expensive resource, you need to open it first (asking for password), use it to do the job quickly and then close (erase both password and secrets in memory). 
They shouldn't be stored as properties of some long-lived objects.